patch_recurrence/dehaze.py

```diff
@@ -71,33 +69,8 @@ def main():
         logger.info("Putting patches in buckets ...")
         steps.set_patch_buckets(patches, constants)
 
-        logger.info("Assigning colors to each bucket ...")
-        bucket_img = tools.set_buckets(img, patches[0], constants)
-
-        logger.info("Displaying smooth buckety image ...")
-        tools.show_buckety_img([img, bucket_img], constants)
-
-        '''
-        logger.info("Generating pairs of patches ...")
-        pairs = steps.generate_pairs(patches, constants)
-
-        print pairs.shape
-        print pairs
-        '''
-
-        sys.exit()
-
-        logger.info("Extracting patches ...")
-        patches = steps.generate_patches(scaled_imgs, constants, False)
-
-        logger.info("Smoothening std deviations of patches ...")
-        steps.smoothen(scaled_imgs, patches, constants)
-
         logger.info("Generating pairs of patches ...")
-        pairs = steps.generate_pairs(patches, constants)
-
-        print pairs.shape
-        print pairs
+        pairs = steps.generate_pairs(scaled_imgs, patches, constants)
```

Comment:
There are a few functions in the pipeline that follows - `steps.remove_duplicates`, `steps.filter_pairs`, `steps.remove_overlaps`, `steps.remove_outliers`. Are all of them needed now? Perhaps we can check the number of pairs after each step, just to make sure.